setup.py

```diff
@@ -1,36 +1,35 @@
+try:
+    from setuptools import setup, find_packages
+except ImportError:
+    from distutils.core import setup
+
 import subprocess
 import sys
-from setuptools import setup, find_packages
 import versioneer
 
-
-# Utility method(s)
-def install(package):
-    subprocess.check_call(
-        [sys.executable, "-m", "pip", "install", "--user", package]
-    )
-
-
 # Requirements definitions
 SETUP_REQUIRES = [
-    "setuptools == 59.5.0",
+    "setuptools==62.3",
 ]
 
 INSTALL_REQUIRES = [
-    "sqlalchemy",
-    "pandas>=1.1.0",
-    "numpy",
-    "timer",
-    "tqdm",
-    "torch-cluster==1.5.9",
+    "sqlalchemy>=1.4.37",
+    "pandas>=1.4",
+    "numpy>=1.22",
+    "timer>=0.2",
+    "tqdm>=4.64",
+    "dill>=0.3",
+    "wandb>=0.12",
+    "matplotlib>=3.5",
+    "scikit_learn~=1.1.1",
+    "scipy>=1.8",
+    "torch~=1.11",
+    "torch-cluster==1.6.0",
     "torch-scatter==2.0.9",
-    "torch-sparse==0.6.12",
+    "torch-sparse==0.6.13",
     "torch-spline-conv==1.2.1",
```

Comment:
I figure we should aim to specify version _ranges_ (e.g., `torch_cluster>=1.6.0` or similar) rather than specific versions, or — ideally — not specify versions here at all, unless it is strictly necessary for compatibility.

Reply:
The specification of version for the pytorch geometric related packages (and pytorch geometric itself) are based on past experience, where the geometric packages are not very flexible.
Further, it seems the torch packages in general often fall into paradigms, where for example torch 1.10 and 1.11 are specifically needed for versions of the torch geometric and cu*** installations.
So the strict specification was more to make sure that the version was always correct.